### PR TITLE
Lookup endpoint in swagger documentation (Task #15805)

### PR DESCRIPTION
**Pull request by acrias**
This PR adds the lookup endpoint in the swagger documentation.

### CODE REVIEW

File: src/Swagger/Annotation.php [hunk 1]
```diff
@@ -153,6 +153,90 @@ class Annotation
                 )
             )
 
+            @SWG\Get(
+                path="/api/{{module_url}}/lookup.json",
```
Comment (georgeconstantinou) [inline src/Swagger/Annotation.php]: @acrias I think we can remove the .json extension here to keep it consistent with the rest of the endpoints.

File: src/Swagger/Annotation.php [hunk 2]
```diff
@@ -153,6 +153,90 @@ class Annotation
...
+                summary="Retrieve a list of records {{module_human_singular}} by name",
+                tags={"{{module_human_plural}}"},
+                produces={"application/json"},
```
Comment (georgeconstantinou) [inline src/Swagger/Annotation.php]: @acrias we can also include the `consumes` parameter:
```
consumes={"application/json"}
```

File: src/Swagger/Annotation.php (hunk 1 above)
Comment (acrias) [reply src/Swagger/Annotation.php]: I think I have tried it too but it does not work

File: src/Swagger/Annotation.php (hunk 2 above)
Comment (acrias) [reply src/Swagger/Annotation.php]: @georgeconstantinou is it needed? because we are not sending json is get and the parameters is on the url

Comment (georgeconstantinou) [reply src/Swagger/Annotation.php]: true, we can skip this.

File: src/Swagger/Annotation.php (hunk 1 above)
Comment (acrias) [reply src/Swagger/Annotation.php]: @georgeconstantinou you mean `/api/things/lookup?query=and&limit=4&page=1` ? It does not work.

Comment (georgeconstantinou) [reply src/Swagger/Annotation.php]: For me it works, do you have a record that includes `and` in the name?

**Comment by asaliev** [comment]
AFAIK the API can also respond with 404 (when debug is disabled) when you request a page which is out of bounds. Not sure if you want to include this as part of the swagger response samples.

File: src/Swagger/Annotation.php (hunk 1 above)
Comment (acrias) [reply src/Swagger/Annotation.php]: Okay it works. Most probably I was only changing the url instead of the actual endpoint inthe annotation file

**Pull request merged**